backend/src/api/comment/create_comment.rs

```diff
@@ -17,11 +19,15 @@ pub(super) async fn post<D: SqliteDatabase>(
     Authorize(user): Authorize,
     Extension(state): Extension<Arc<AppState<D>>>,
 ) -> Result<Json<Comment>, Error> {
-    let Json(text) = request?;
+    let Json(mut text) = request?;
     let conn = state.pool.get().await.context("Failed to get connection")?;
 
     let now = SystemTime::UNIX_EPOCH.elapsed().unwrap().as_secs();
 
+    // Chek for possible aliases within the text
+    // and return the modified string
+    text = extract_alias(text, &state).await?;
+
     conn.interact(move |conn| {
         if !image::image_exists(&image_key, conn)? {
```

Comment:
If this is checked before replacing aliases we would have to do less work if an invalid image key is sent in, it's not super important to change though since I don't expect this error to be hit often.